torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -292,27 +277,24 @@ bool Reducer::ddp_graph_static() {
 }
 
 void Reducer::initialize_local_used_map() {
-  const auto replica_count = replicas_.size();
-  const auto variable_count = replicas_[0].size();
-  local_used_maps_.resize(replica_count);
-  local_used_maps_dev_.resize(replica_count);
-
-  for (const auto i : c10::irange(replica_count)) {
-    at::TensorOptions options;
-    options = options.dtype(at::kInt);
-
-    // Deliberately don't pin the memory even if local_used_maps_dev_ will
-    // be cuda. See Note [local_used_maps_ -> local_used_maps_dev copying]
-    local_used_maps_[i] =
-        at::zeros({static_cast<long>(variable_count)}, options);
-
-    // This tensor needs to be on the same device as replica because backend
-    // such as NCCL may not support CPU tensors, and hence it might not work
-    // if we always put it on CPU.
-    options = options.device(replicas_[i][0].device());
-    local_used_maps_dev_[i] =
-        at::empty({static_cast<long>(variable_count)}, options);
-  }
+  const auto variable_count = params_.size();
+  local_used_maps_.resize(1);
+  local_used_maps_dev_.resize(1);
```

Comment:
could we define 'at::Tensor local_used_maps_' and 'at::Tensor local_used_maps_dev_'?

Reply:
The name needs to be changed to `at::Tensor local_used_map_` and `at::Tensor local_used_map_dev_`? (maps -> map)

Reply:
If I change the typing of these variables, I will also have to change their usages. Just to be clear, is that what you're asking for?

Reply:
Addressed in https://github.com/pytorch/pytorch/pull/65380